### PR TITLE
Send email notification with smtplib.

### PR DESCRIPTION
**Pull request by adrianmoloce**
- Send email notification if send_mail flag is True
- Created config file for environment variables
- Created unit tests for scenarios where send email is successful and when sending it encounters errors.
- Add python-dotenv and pytest-mock as dependencies.

### CODE REVIEW

File: src/invoice_utils/config.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+MAIL_HOST = os.getenv("MAIL_HOST")
```
Comment (koosie0507) [inline src/invoice_utils/config.py]: It's always nice to namespace the names of the env vars with your app name. The global var's name can remain the same, but the env var should be namespaced. You can use pretty much anything as the namespace as long as there isn't a high chance of name collision. Longer strings have lower chances of colliding. E.g. `MAIL_HOST` -> `INVOICE_UTILS_MAIL_HOST`

File: src/invoice_utils/config.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+MAIL_PORT = os.getenv("MAIL_PORT")
```
Comment (koosie0507) [inline src/invoice_utils/config.py]: `os.getenv` allows returning default values if the env var isn't present. Might want to use this feature for some of these variables.

File: tests/test_web.py [hunk 3]
```diff
@@ -26,21 +31,51 @@ def test_send_mail_param_not_provided(http, caplog, invoice_request_body):
...
 
-def test_send_mail_param_is_true_and_address_provided(http, caplog, invoice_request_body):
+def test_send_mail_param_is_true_and_address_provided(
+    http, caplog, invoice_request_body, mocker
+):
     invoice_request_body["send_mail"] = True
     invoice_request_body["address"] = "test@email.com"
+
+    mock_smtp = mocker.MagicMock(name="invoice_utils.web.smtplib.SMTP")
+    mocker.patch("invoice_utils.web.smtplib.SMTP", new=mock_smtp)
```
Comment (koosie0507) [inline tests/test_web.py]: Does `invoice_utils.web.smtplib.SMTP` get un-patched at the end of running the test?

File: tests/test_web.py [hunk 4]
```diff
@@ -26,21 +31,51 @@ def test_send_mail_param_not_provided(http, caplog, invoice_request_body):
...
+    )
     assert res.status_code == 422
 
 
-def test_send_mail_param_is_true_and_address_provided(http, caplog, invoice_request_body):
+def test_send_mail_param_is_true_and_address_provided(
+    http, caplog, invoice_request_body, mocker
+):
     invoice_request_body["send_mail"] = True
     invoice_request_body["address"] = "test@email.com"
```
Comment (koosie0507) [inline tests/test_web.py]: Remind me to talk to you about the [4 rules of simple design](https://martinfowler.com/bliki/BeckDesignRules.html) next time we meet (online or in person).

File: tests/test_web.py [hunk 5]
```diff
@@ -26,21 +31,51 @@ def test_send_mail_param_not_provided(http, caplog, invoice_request_body):
...
     with caplog.at_level("INFO"):
         res = http.post("/invoice", json=invoice_request_body)
 
     assert "Report was sent to test@email.com" in caplog.messages
+    # Because it's used as a context manager.
+    assert mock_smtp.return_value.__enter__.return_value.sendmail.call_count == 1
     assert res.status_code == 201
+
+
+def test_send_mail_fails_for_smtp_exception(http, caplog, invoice_request_body, mocker):
```
Comment (koosie0507) [inline tests/test_web.py]: Coupled with the commit structure, this doesn't look like the outcome of TDD. Should've had one assertion per test and should've started to extract common setup behaviour for the multiple tests that use it.
Next time, try limiting yourself to only one assertion per test (that fails)

File: tests/test_web.py (hunk 3 above)
Comment (adrianmoloce) [reply tests/test_web.py]: It's a feature of pytest-mock, it automatically undoing the mocking at the end of the test.

File: src/invoice_utils/config.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+DEFAULT_MAIL_HOST = "smtp.gmail.com"
+DEFAULT_PORT = 587
+
+INVOICE_UTILS_MAIL_HOST = os.getenv("INVOICE_UTILS_MAIL_HOST", DEFAULT_MAIL_HOST)
+INVOICE_UTILS_MAIL_PORT = os.getenv("INVOICE_UTILS_MAIL_PORT", DEFAULT_PORT)
+INVOICE_UTILS_MAIL_LOGIN_USER = os.getenv("INVOICE_UTILS_MAIL_LOGIN_USER")
+INVOICE_UTILS_MAIL_LOGIN_PASSWORD = os.getenv("INVOICE_UTILS_MAIL_LOGIN_PASSWORD")
+INVOICE_UTILS_SENDER_EMAIL = os.getenv("INVOICE_UTILS_SENDER_EMAIL")
+INVOICE_UTILS_MAIL_SUBJECT = "Invoice generated with invoice-utils"
```
Comment (koosie0507) [inline src/invoice_utils/config.py]: `os.environ["INVOICE_UTILS_MAIL_SUBJECT"] = ""`

**Pull request merged**